### PR TITLE
Feature/genius 2034 access private and trusted keys in their normal form

### PR DESCRIPTION
**Pull request by schroenser**
I took the liberty to extract this from the applications so the code remains consistent.

### CODE REVIEW

**Comment by lesmou** [comment]
Added @JanDoerrenhaus since @bannmann is on holiday.

File: core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            result = (RSAPrivateKey) keyFactory.generatePrivate(keySpec);
+        }
+        catch (InvalidKeySpecException e)
+        {
+            throw new ConfigurationException("Invalid key specification", e);
+        }
+        return result;
+    }
+
+    public RSAPublicKey publicKeyFromX509(String src)
```
Comment (JanDoerrenhaus) [inline core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java:48]: Public before private.

File: core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            result = (RSAPublicKey) keyFactory.generatePublic(keySpec);
+        }
+        catch (InvalidKeySpecException e)
+        {
+            throw new ConfigurationException("Invalid key specification", e);
+        }
+        return result;
+    }
+
+    private static String stripHeaderFooterAndLineBreaks(String src)
```
Comment (JanDoerrenhaus) [inline core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java:78]: Static before non-static.

File: core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.regex.Pattern;
+
+import lombok.experimental.UtilityClass;
+
+import com.github.mizool.core.exception.ConfigurationException;
+
+@UtilityClass
+public class RsaKeys
+{
+    private static final Pattern HEADER_FOOTER_AND_LINE_BREAKS = Pattern.compile(
```
Comment (JanDoerrenhaus) [inline core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java:23]: Oh, right. This is a @UtilityClass. Then remove the static here as well.

Comment (schroenser) [reply core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java:23]: I already had the same idea. However, when I removed the keyword, Sonar started to complain that the result of `Pattern.compile` should always be assigned to a `static final` field. Looks like the sonar rule does not know about Lombok.
I'm afraid that is a conflict of interest with [OMG-29](https://incub8.atlassian.net/browse/OMG-29). I'm not sure which route we should take with this.

I think, I'll remove the keyword here, giving our decision in the OMG precedence, we mark this as false positive in Sonar and talk about this during one of our next meetings.

Comment (schroenser) [reply core/src/main/java/com/github/mizool/core/rsa/RsaKeys.java]: I've created [CCCP-34](https://incub8.atlassian.net/browse/CCCP-34) to further this discussion.

**Pull request merged**